sdk/iot/provisioning/src/az_iot_provisioning_client.c

```diff
@@ -167,7 +170,80 @@ AZ_NODISCARD az_result az_iot_provisioning_client_register_get_subscribe_topic_f
   return AZ_OK;
 }
 
-AZ_INLINE az_iot_provisioning_client_registration_state registration_state_default()
+// $dps/registrations/PUT/iotdps-register/?$rid=%s
+AZ_NODISCARD az_result az_iot_provisioning_client_register_get_publish_topic(
+    az_iot_provisioning_client const* client,
+    char* mqtt_topic,
+    size_t mqtt_topic_size,
+    size_t* out_mqtt_topic_length)
+{
+  (void)client;
+
+  AZ_PRECONDITION_NOT_NULL(client);
+  AZ_PRECONDITION_NOT_NULL(mqtt_topic);
+  AZ_PRECONDITION(mqtt_topic_size > 0);
+
+  az_span mqtt_topic_span = az_span_init((uint8_t*)mqtt_topic, (int32_t)mqtt_topic_size);
+  az_span str_dps_registrations = _az_iot_provisioning_get_str_dps_registrations();
+
+  int32_t required_length
+      = az_span_size(str_dps_registrations) + az_span_size(str_put_iotdps_register);
+
+  AZ_RETURN_IF_NOT_ENOUGH_SIZE(mqtt_topic_span, required_length + (int32_t)sizeof((uint8_t)'\0'));
+
+  az_span remainder = az_span_copy(mqtt_topic_span, str_dps_registrations);
+  remainder = az_span_copy(remainder, str_put_iotdps_register);
+  remainder = az_span_copy_u8(remainder, '\0');
+
+  if (out_mqtt_topic_length)
+  {
+    *out_mqtt_topic_length = (size_t)required_length;
+  }
+
+  return AZ_OK;
+}
+
+// Topic: $dps/registrations/GET/iotdps-get-operationstatus/?$rid=%s&operationId=%s
+AZ_NODISCARD az_result az_iot_provisioning_client_query_status_get_publish_topic(
+    az_iot_provisioning_client const* client,
+    az_iot_provisioning_client_register_response const* register_response,
+    char* mqtt_topic,
+    size_t mqtt_topic_size,
+    size_t* out_mqtt_topic_length)
+{
+  (void)client;
+
+  AZ_PRECONDITION_NOT_NULL(client);
+  AZ_PRECONDITION_NOT_NULL(mqtt_topic);
+  AZ_PRECONDITION(mqtt_topic_size > 0);
+
+  AZ_PRECONDITION_NOT_NULL(register_response);
+  AZ_PRECONDITION_VALID_SPAN(register_response->operation_id, 1, false);
+
+  az_span mqtt_topic_span = az_span_init((uint8_t*)mqtt_topic, (int32_t)mqtt_topic_size);
+  az_span str_dps_registrations = _az_iot_provisioning_get_str_dps_registrations();
+
+  int32_t required_length = az_span_size(str_dps_registrations)
+      + az_span_size(str_get_iotdps_get_operationstatus)
+      + az_span_size(register_response->operation_id);
+
+  AZ_RETURN_IF_NOT_ENOUGH_SIZE(mqtt_topic_span, required_length + (int32_t)sizeof((uint8_t)'\0'));
```

Comment:
just as a note here...no action items for you. But I have a bug task open to go through and consolidate all the commonly used `const`'s like the null terminator, question mark, etc into one of the common files. So this will in the future be changed to something like 
```c
.. + sizeof(null_terminator)
```
or something

Reply:
I don't like hiding single characters behind named constants. '#' should not be `hash_tag`. It just makes it harder to read - now someone new to the code needs to go to some obscure top-level header file to figure out what otherwise is a single character.

The only benefit to define constants is to avoid spreading common values _that may change_ through the code. The null terminator will never change. Same thing goes for `'#'`.